zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -210,6 +211,18 @@ private void initInterpreterSetting(InterpreterSetting interpreterSetting) {
         .setRecoveryStorage(recoveryStorage)
         .setInterpreterEventServer(interpreterEventServer)
         .postProcessing();
+    
+    // Since the ClusterInterpreterLauncher is lazy, dynamically generated, So in cluster mode,
+    // when the zeppelin service starts, Create a ClusterInterpreterLauncher object,
+    // This allows the ClusterInterpreterLauncher to listen for cluster events.
+    if (conf.isClusterMode()) {
+      try {
+        String pluginName = interpreterSetting.getLauncherPlugin();
+        PluginManager.get().loadInterpreterLauncher(pluginName, recoveryStorage);
```

Comment:
Why init IntepreterLauncher here instead of InterpreterSetting ?

Reply:
Modify `ClusterInterpreterLauncher` initialization in zeppelin server.